scripts/lib/CIME/XML/env_batch.py

```diff
@@ -95,51 +94,47 @@ def get_type_info(self, vid):
         return type_info
 
     def get_jobs(self):
-        groups = self.get_nodes("group")
+        groups = self.get_children("group")
         results = []
         for group in groups:
-            if group.get("id") not in ["job_submission", "config_batch"]:
-                results.append(group.get("id"))
+            if self.get(group, "id") not in ["job_submission", "config_batch"]:
+                results.append(self.get(group, "id"))
 
         return results
 
     def create_job_groups(self, batch_jobs):
         # Subtle: in order to support dynamic batch jobs, we need to remove the
         # job_submission group and replace with job-based groups
 
-        orig_group = self.get_optional_node("group", {"id":"job_submission"})
-        expect(orig_group, "Looks like job groups have already been created")
+        orig_group = self.get_child("group", {"id":"job_submission"},
```

Comment:
even though it looks different, the semantics of this method was not changed at all.